starlette/staticfiles.py

```diff
@@ -146,17 +146,18 @@ async def get_response(self, path: str, scope: Scope) -> Response:
                 return self.file_response(full_path, stat_result, scope)
 
         if self.html:
-            # Check for '404.html' if we're in HTML mode.
-            full_path, stat_result = await anyio.to_thread.run_sync(
-                self.lookup_path, "404.html"
-            )
-            if stat_result and stat.S_ISREG(stat_result.st_mode):
-                return FileResponse(
-                    full_path,
-                    stat_result=stat_result,
-                    method=scope["method"],
-                    status_code=404,
+            # Check for '200.html' and '404.html' if we're in HTML mode.
+            for status in [200, 404]:
```

Comment:
@graingert @aminalaee any ideas to not do 2 lookups for 404s?

Reply:
The thinking behind checking for 200.html first and 404.html later, was that handling 200.html should be faster than 404.html. Because serving 200.html is a regular (non-error) operation, the second is an exception.

Reply:
It depends. IMO, most users do not need 200.html, 404.html will be used. So checking for 404 first is a more appropriate solution for the library.

Reply:
Alternatively, the logic inside the `if self.html:` in a method, and then people can create their own `StaticFiles` class.

e.g.:
```py
class MyStaticFiles(StaticFiles):
   def html_method(self):
       return FileResponse(200)
```